server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -523,6 +525,20 @@ public String getHeader(String header) {
         return headers.get(header);
     }
 
+    public void addQueryGroupHeadersTo(final ThreadContext threadContext) {
+        // For now this header will be coming from HTTP headers but in second phase this header
+
+        // We will use this constant from QueryGroup Service once the framework changes are done
+        final String QUERY_GROUP_ID_HEADER = "queryGroupId";
+        final String requestQueryGroupId = threadContext.getHeader(QUERY_GROUP_ID_HEADER);
```

Comment:
I think we should do some sanity checks before adding the qg id to the header. 
Like the below cases are invalid
- empty string
- blank empty space character
- and ability to add more ?

Reply:
I am planning to keep this clean since we will do this due diligence at the place where we will resolve the query group for the request.